internal/provider/models/alerts/threshold.go

```diff
@@ -0,0 +1,187 @@
+	EventType         StringValue `tfsdk:"event_type"`
+	GroupBy           ListValue   `tfsdk:"group_by" user_config:"true"`
+	Operation         StringValue `tfsdk:"operation" user_config:"true"`
+	Conditional       ObjectValue `tfsdk:"conditional" user_config:"true"`
+	WindowType        StringValue `tfsdk:"window_type" user_config:"true"`
+	WindowDurationMin Int64Value  `tfsdk:"window_duration_minutes" user_config:"true"`
+	Script            StringValue `tfsdk:"script" user_config:"true"`
+	EventTimestamp    StringValue `tfsdk:"event_timestamp" user_config:"true"`
+	Severity          StringValue `tfsdk:"severity" user_config:"true"`
+	Style             StringValue `tfsdk:"style" user_config:"true"`
+	Subject           StringValue `tfsdk:"subject" user_config:"true"`
+	Message           StringValue `tfsdk:"message" user_config:"true"`
+	IngestionKey      StringValue `tfsdk:"ingestion_key" user_config:"true"`
+}
+
+var ThresholdAlertResourceSchema = schema.Schema{
+	Description: "Represents a Threshold Alert in a Pipeline",
+	Attributes: ExtendBaseAttributes(map[string]schema.Attribute{
+		"conditional": schema.SingleNestedAttribute{
+			Required:    true, // ParentConditionalAttribute is not required by default
+			Description: ParentConditionalAttribute(Non_Change_Operator_Labels).Description,
+			Attributes:  ParentConditionalAttribute(Non_Change_Operator_Labels).Attributes,
+		},
+	}),
+}
+
+// From terraform schema/model to a struct for sending to the API
+func ThresholdAlertFromModel(plan *ThresholdAlertModel, previousState *ThresholdAlertModel) (*Alert, diag.Diagnostics) {
+	dd := diag.Diagnostics{}
+
+	CustomErrorChecks(&CheckedFields{
+		Operation:      plan.Operation,
+		EventType:      plan.EventType,
+		Script:         plan.Script,
+		EventTimestamp: plan.EventTimestamp,
+		GroupBy:        plan.GroupBy,
+	}, &dd)
+
+	if dd.HasError() {
+		return nil, dd
+	}
+
+	// Inputs are required, so we can assemble them first
+	inputs := make([]string, 0)
+	for _, v := range plan.Inputs.Elements() {
+		value, _ := v.(StringValue)
+		inputs = append(inputs, value.ValueString())
+	}
+
+	component := Alert{
+		PipelineId:    plan.PipelineId.ValueString(),
+		ComponentKind: plan.ComponentKind.ValueString(),
+		ComponentId:   plan.ComponentId.ValueString(),
+		Inputs:        inputs,
+		AlertConfig: map[string]any{
+			"general": map[string]any{
+				"name": plan.Name.ValueString(),
+			},
+			"evaluation": map[string]any{
+				"alert_type":  ALERT_TYPE_THRESHOLD, // Required for the API, but hidden from the user here
+				"event_type":  plan.EventType.ValueString(),
+				"operation":   Aggregate_Operations[plan.Operation.ValueString()],
+				"conditional": UnwindConditionalFromModel(plan.Conditional),
+			},
+			"alert_payload": map[string]any{
+				"subject": plan.Subject.ValueString(),
+				"message": plan.Message.ValueString(),
+				"destination": map[string]any{
+					"ingestion_key": plan.IngestionKey.ValueString(),
+				},
+			},
+		},
+	}
```

Comment:
question: i'm trying to understand the reasoning here, why dont we make this staticly typed instead of using the empty interface?

Reply:
For the `alert_payload` map, it could be `map[string]string` because all the values are strings. More often than not, these subobjects contain numbers, lists or objects, which requires us to declare the field type as `any`. Not sure it really provides much safety in this specific context.

Even if we do statically type this map, the unmarshal data is going to be type `map[string]any` since the library function can't determine the type until the decoding happens.

Reply:
And to tie up this comment--here, the types of the values are mixed.  In the case of `alert_payload`, `destination` is a map, so you can't use `map[string]string` for alert payload.